src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1063,8 +1098,33 @@ private Conversion ClassifyImplicitBuiltInConversionFromExpression(BoundExpressi
             return Conversion.NoConversion;
         }
 
-        private Conversion GetSwitchExpressionConversion(BoundExpression source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        private Conversion ClassifyImplicitUtf8StringLiteralConversion(BoundExpression sourceExpression, CSharpCompilation compilation, TypeSymbol destination)
+        {
+            // The language will allow conversions between string constants and byte sequences
+            // where the text is converted into the equivalent UTF8 byte representation.
+            // Specifically the compiler will allow for implicit conversions from string constants to
+            // byte[], Span<byte>, and ReadOnlySpan<byte>.
+            ConstantValue constantValue = sourceExpression.ConstantValue;
+
+            TypeSymbol destinationOriginalDefinition = destination.OriginalDefinition;
+            if (constantValue?.IsString == true && // PROTOTYPE(UTF8StringLiterals) : confirm if we actually want it to work with 'null' constant value.
+                sourceExpression.Type?.SpecialType == SpecialType.System_String &&
+                (destination is ArrayTypeSymbol { IsSZArray: true, ElementType.SpecialType: SpecialType.System_Byte } || // byte[]
+                 ((destinationOriginalDefinition.Equals(compilation.GetWellKnownType(WellKnownType.System_Span_T), TypeCompareKind.AllIgnoreOptions) ||             // Span<T>
+                   destinationOriginalDefinition.Equals(compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions)) &&    // ReadOnlySpan<T>
+                   destinationOriginalDefinition.TypeKind == TypeKind.Struct && destinationOriginalDefinition.IsRefLikeType &&
```

Comment:
>destinationOriginalDefinition.TypeKind == TypeKind.Struct && destinationOriginalDefinition.IsRefLikeType && [](http://example.com/codeflow?start=19&length=107)

Consider moving these checks ahead of the calls to `Equals(compilation.GetWellKnownType(...))` since these seem less expensive.